tests/Main/DateOnlyTests.fs

```diff
@@ -0,0 +1,93 @@
+            let date = DateOnly (1, 1, 2)
+            date.Day |> equal 2
+            let date = DateOnly (2000, 2, 28)
+            date.Day |> equal 28
+            let date = DateOnly.MaxValue
+            date.Day |> equal 31
+            let date = DateOnly.MinValue
+            date.Day |> equal 1
+
+        testCase "DateOnly Month works" <| fun () ->
+            let date = DateOnly (1, 1, 2)
+            date.Month |> equal 1
+            let date = DateOnly (2000, 2, 28)
+            date.Month |> equal 2
+            let date = DateOnly.MaxValue
+            date.Month |> equal 12
+            let date = DateOnly.MinValue
+            date.Month |> equal 1
+
+        testCase "DateOnly Year works" <| fun () ->
+            let date = DateOnly (1, 1, 2)
+            date.Year |> equal 1
+            let date = DateOnly (2000, 2, 28)
+            date.Year |> equal 2000
+            let date = DateOnly.MaxValue
+            date.Year |> equal 9999
+            let date = DateOnly.MinValue
+            date.Year |> equal 1
+
+        testCase "DateOnly DayOfWeek works" <| fun () ->
+            let date = DateOnly (1, 1, 2)
+            date.DayOfWeek |> equal DayOfWeek.Tuesday
+            let date = DateOnly (2000, 2, 28)
+            date.DayOfWeek |> equal DayOfWeek.Monday
+            let date = DateOnly.MaxValue
+            date.DayOfWeek |> equal DayOfWeek.Friday
+            let date = DateOnly.MinValue
+            date.DayOfWeek |> equal DayOfWeek.Monday
+
+        testCase "DateOnly DayOfYear works" <| fun () ->
+            let date = DateOnly (1, 1, 2)
+            date.DayOfYear |> equal 2
+            let date = DateOnly (2000, 2, 28)
+            date.DayOfYear |> equal 59
+            let date = DateOnly.MaxValue
+            date.DayOfYear |> equal 365
+            let date = DateOnly.MinValue
+            date.DayOfYear |> equal 1
+
+        testCase "DateOnly FromDateTime works" <| fun () ->
+            let dateTime = DateTime (2000, 2, 27, 23, 30, 0, DateTimeKind.Local)
+            DateOnly.FromDateTime dateTime |> equal (DateOnly (2000, 2, 27))
+            let dateTime = DateTime (2000, 2, 27, 23, 30, 0, DateTimeKind.Utc)
+            DateOnly.FromDateTime dateTime |> equal (DateOnly (2000, 2, 27))
+            let dateTime = DateTime (2000, 2, 28, 0, 30, 0, DateTimeKind.Local)
+            DateOnly.FromDateTime dateTime |> equal (DateOnly (2000, 2, 28))
+            let dateTime = DateTime (2000, 2, 28, 0, 30, 0, DateTimeKind.Utc)
+            DateOnly.FromDateTime dateTime |> equal (DateOnly (2000, 2, 28))
+            let dateTime = DateTime.Now
+            DateOnly.FromDateTime dateTime |> equal (DateOnly (DateTime.Now.Year, DateTime.Now.Month, DateTime.Now.Day))
+            //let dateTime = DateTime.MaxValue
+            //DateOnly.FromDateTime dateTime |> equal DateOnly.MaxValue
+            //let dateTime = DateTime.MinValue
+            //DateOnly.FromDateTime dateTime |> equal DateOnly.MinValue
```

Comment:
These tests don't work because DateTime.MinValue/MaxValue is being created with `DateKind.Unspecified` and not UTC, so when you call `getTicks` on it, you get a different number back than what it was created with. Similarly, I also think there is a bug in `create` of `Date.ts` where `setFullYear` is called instead of `setUTCFullYear` when the requested kind is UTC.

Reply:
It's not so simple and this scenario is a corner case with little practical use. Let's forget it.